tests/conftest.py

```diff
@@ -174,3 +176,44 @@ def async_authenticated_client(api_base_url, token_url, client_id, client_secret
 @pytest.fixture
 def access_token():
     return settings.ACCESS_TOKEN
+
+
+@pytest.fixture
+def fake_key_file(tmp_path_factory):
+    tmp_path_factory._basetemp = Path().home()
+    key_path = tmp_path_factory.mktemp(".sfapi_test1", numbered=False) / "key.pem"
+
+    # Make a fake key for testing
+    key_path.write_text(
+        f"""abcdefghijlmo
+-----BEGIN RSA PRIVATE KEY-----
+{rsa.generate_private_key(public_exponent=65537, key_size=2048)}
+-----END RSA PRIVATE KEY-----
+"""
+    )
+    key_path.chmod(0o100600)
+    try:
+        yield key_path
+    finally:
+        # make sure to cleanup the test since we put a file in ~/.sfapi_test
+        temp_path = Path().home() / ".sfapi_test1"
+        if temp_path.exists():
+            (temp_path / "key.pem").unlink(missing_ok=True)
+            temp_path.rmdir()
```

Comment:
Sorry all the operations that could raise an exception should be in the try block, so like this:

```python
   try:
       tmp_path_factory._basetemp = Path().home()
       key_path = tmp_path_factory.mktemp(".sfapi_test1", numbered=False) / "key.pem"

        # Make a fake key for testing
        key_path.write_text(
        f"""abcdefghijlmo
-----BEGIN RSA PRIVATE KEY-----
{rsa.generate_private_key(public_exponent=65537, key_size=2048)}
-----END RSA PRIVATE KEY-----
"""
        )
        key_path.chmod(0o100600)
        
        yield key_path
    finally:
        # make sure to cleanup the test since we put a file in ~/.sfapi_test
        temp_path = Path().home() / ".sfapi_test1"
        if temp_path.exists():
            (temp_path / "key.pem").unlink(missing_ok=True)
            temp_path.rmdir()
```

Reply:
Ah okay makes sense!